test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs

```diff
@@ -0,0 +1,180 @@
+            GetFloorPlan(),
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            new Point[] { Point.ByCoordinates(100, 100, 0) },
+            true);
+
+         Assert.NotNull(pathOfTravelManyToMany);
+         Assert.Equals(pathOfTravelManyToMany.GetLength(0), 1);
+         Assert.NotNull(pathOfTravelManyToMany[0]);
+      }
+
+      [Test]
+      [TestModel(@".\Empty.rvt")]
+      public void Create_NullView()
+      {
+         Assert.Throws(
+            typeof(System.ArgumentNullException),
+            () => PathOfTravel.ByFloorPlanPoints(
+               null,                                                 // null view
+               new Point[] { Point.ByCoordinates(0, 0, 0) },
+               new Point[] { Point.ByCoordinates(100, 100, 0) },
+               false));
+
+         Assert.Throws(
+            typeof(System.ArgumentNullException),
+            () => PathOfTravel.ByFloorPlanPoints(
+               null,                                                 // null view
+               new Point[] { Point.ByCoordinates(0, 0, 0) },
+               new Point[] { Point.ByCoordinates(100, 100, 0) },
+               true));
+      }
+
+      [Test]
+      [TestModel(@".\Empty.rvt")]
+      public void Create_NullStartPointArray()
+      {
+         Assert.Throws(
+            typeof(System.ArgumentNullException),
+            () => PathOfTravel.ByFloorPlanPoints(
+               GetFloorPlan(),                                       // null start point array
+               null,
+               new Point[] { Point.ByCoordinates(100, 100, 0) },
+               false));
+
+         Assert.Throws(
+            typeof(System.ArgumentNullException),
+            () => PathOfTravel.ByFloorPlanPoints(
+               GetFloorPlan(),                                       // null start point array
+               null,
+               new Point[] { Point.ByCoordinates(100, 100, 0) },
+               true));
+      }
+
+      [Test]
+      [TestModel(@".\Empty.rvt")]
+      public void Create_InvalidEndPoint()
+      {
+         Assert.Throws(
+            typeof(System.ArgumentNullException),
+            () => PathOfTravel.ByFloorPlanPoints(
+               GetFloorPlan(),
+               new Point[] { Point.ByCoordinates(0, 0, 0) },
+               null,                                                 // null end point array
+               false));
+
+         Assert.Throws(
+            typeof(System.ArgumentNullException),
+            () => PathOfTravel.ByFloorPlanPoints(
+               GetFloorPlan(),
+               new Point[] { Point.ByCoordinates(0, 0, 0) },
+               null,                                                 // null end point array
+               true));
+      }
+
+      [Test]
+      [TestModel(@".\Empty.rvt")]
+      public void Create_DifferentPointCounts()
+      {
+         Assert.Throws(
+            typeof(System.ArgumentException),
```

Comment:
Is ArgumentException should be ArgumentNullException?

Reply:
No, the arguments are valid objects, just a different number of points in each array.

Reply:
But your startPtList has no element, a ArgumentNullException would throw first here https://github.com/DynamoDS/DynamoRevit/blob/534b165555ab473ab7a4a1f08a900c33ebce8820/src/Libraries/RevitNodes/Elements/PathOfTravel.cs#L77 .
I think it couldn't get a ArgumentExcetion in this test.